typescript/packages/subsurface-viewer/src/layers/axes2d/axes2DLayer.ts

```diff
@@ -144,8 +148,22 @@ const fontInfo = {
 };
 
 export default class Axes2DLayer extends Layer<Axes2DLayerProps> {
-    shouldUpdateState({ changeFlags }: UpdateParameters<this>) {
-        return changeFlags.viewportChanged;
+    shouldUpdateState({
+        props,
+        oldProps,
+        context,
+        changeFlags,
+    }: UpdateParameters<this>): boolean {
+        return (
+            super.shouldUpdateState({
+                props,
+                oldProps,
+                context,
+                changeFlags,
+            }) ||
+            changeFlags.propsOrDataChanged ||
```

Comment:
Probably useless (there is no data and super.shouldUpdateState should handle props).

BTW, we should be consistent with other layers (in particular 3D axes layer)

Reply:
Yes that's true. They are the same now.